lib/ruby_lsp/listeners/completion.rb

```diff
@@ -406,8 +408,8 @@ def build_entry_completion(real_name, incomplete_name, range, entries, top_level
       # ```
       sig { params(entry_name: String).returns(T::Boolean) }
       def top_level?(entry_name)
-        @nesting.length.downto(0).each do |i|
-          prefix = T.must(@nesting[0...i]).join("::")
+        @target_context.nesting.length.downto(0).each do |i|
+          prefix = T.must(@target_context.nesting[0...i]).join("::")
```

Comment:
This is invoked in a loop, so let's avoid the double method dispatch.
```suggestion
        nesting = @target_context.nesting
        nesting.length.downto(0).each do |i|
          prefix = T.must(nesting[0...i]).join("::")
```